lib/salus/scanners/report_pom_xml.rb

```diff
@@ -0,0 +1,47 @@
+require 'salus/scanners/base'
+
+# Report Java library usage
+
+module Salus::Scanners
+  class ReportPomXml < Base
+    def run
+      shell_return = run_shell("bin/parse_pom_xml #{@repository.path_to_repo}/pom.xml", chdir: nil)
+
+      if !shell_return.success?
+        report_error(shell_return.stderr)
+        return
+      end
+
+      dependencies = nil
+      begin
+        dependencies = JSON.parse(shell_return.stdout)
+      rescue JSON::ParserError
+        err_msg = "Could not parse JSON returned by bin/parse_pom_xml's stdout!"
+        report_stderr(err_msg)
+        report_error(err_msg)
+      end
+
+      dependencies.each do |dependency|
+        group_id = dependency['group_id']
+        artifact_id = dependency['artifact_id']
+        version = dependency['version']
+        report_error('No group ID found for a dependency!') if group_id.nil?
+        report_error('No artifact ID found for a dependency!') if artifact_id.nil?
+        report_dependency(
+          'pom.xml',
+          type: 'maven',
+          name: "#{group_id}/#{artifact_id}",
+          version: version.nil? ? 'unknown' : version
```

Comment:
We don't need to create the variable 'version' if it is used only once in the code.

Can we use a constant UNKNOWN_VERSION instead?

`version: version.nil? ? UNKNOWN_VERSION : dependency['version']`